sbin/send_notification.py

```diff
@@ -0,0 +1,335 @@
+    course_config_path = os.path.join(
+        COURSE_DIR_PATH, term, course, 'config', 'config.json')
+    course_name = course.strip().upper()
+
+    with open(course_config_path, 'r', encoding="utf-8") as f:
+        data = json.load(f)
+
+        if 'course_name' in data['course_details']:
+            full_name = data['course_details']['course_name'].strip()
+
+            if len(full_name) > 0:
+                course_name += ": " + full_name
+
+    return course_name
+
+
+def connect_db(db_name):
+    """Set up a connection with the specific database."""
+    if os.path.isdir(DB_HOST):
+        connection = (f"postgresql://{DB_USER}:{DB_PASSWORD}@/{db_name}"
+                      f"?host={DB_HOST}")
+    else:
+        connection = (f"postgresql://{DB_USER}:{DB_PASSWORD}@{DB_HOST}"
+                      f"/{db_name}")
+
+    engine = create_engine(connection)
+    db = Session(engine.connect())
+
+    return db
+
+
+def construct_notifications(term, course, pending):
+    """Construct pending gradeable notifications for the current course."""
+    timestamps = {}
+    gradeables, site, email = [], [], []
+    course_name = get_full_course_name(term, course)
+
+    for notification in pending:
+        gradeable = {
+            "id": notification[0],
+            "title": notification[1],
+            "team_id": notification[2],
+            "user_id": notification[3],
+            "user_email": notification[4],
+            "site_enabled": notification[5],
+            "email_enabled": notification[6]
+        }
+
+        timestamp = timestamps.setdefault(
+            gradeable['id'], datetime.datetime.now()
+        )
+
+        # Metadata-related content
+        gradeable_url = (f"{BASE_URL_PATH}/courses/{term}/{course}"
+                         f"/gradeable/{gradeable['id']}")
+        metadata = json.dumps({"url": gradeable_url})
+
+        # Notification-related content
+        notification_content = "Grade Released: " + gradeable["title"]
+        email_subject = (f"[Submitty {course}] Grade Released: "
+                         f"{gradeable['title']}")
+        email_body = (f"An Instructor has released scores in:\n{course_name}"
+                      f"\n\nScores are now available for {gradeable['title']}."
+                      f"\n\nAuthor: System\nClick here for more info: "
+                      f"{gradeable_url}\n\n--\nNOTE: This is an automated "
+                      "email notification, which is unable to receive replies."
+                      "\nPlease refer to the course syllabus for contact "
+                      "information for your teaching staff.\nUpdate your "
+                      "email notification settings for this course here: "
+                      f"{BASE_URL_PATH}/courses/{term}/{course}/notifications"
+                      "/settings"
+                      )
+
+        # Truncate the notification content if it exceeds 40 characters
+        if len(notification_content) > 40:
```

Comment:
I don't think this is an appropriate place to be truncating the notification content.
why 40?  I assume this logic is duplicated within the code in other notification places as well.
But I can't find it by quickly grepping.  If it indeed is duplicate logic
It should be truncated at notification display.  To ease long term maintenance.